File: 11기 코딩테스트_230308/4회차 코딩테스트/FirstAppearTwice/FirstAppearTwice_test.js
import assert from 'assert'
import solution from './FirstAppearTwice.js'

describe('Custom Test', function () {
  it('test 1', function (done) {
    // given
    const s = 'google'

    // when
    const result = solution(s)

    // then
    assert.deepStrictEqual('o', result)
    done();
  });

  it('test 2', function (done) {
    // given
    const s = 'naver'

    // when
    const result = solution(s)

    // then
    assert.deepStrictEqual('', result)
    done();
  });

  it('test 3', function (done) {
    // given
    const s = 'abcdefghijklmnopqrstuvwxyzz'

    // when
    const result = solution(s)

    // then
    assert.deepStrictEqual('z', result)
    done();
  });

  it('test 4', function (done) {
    // given
    const s = 'aabcdefghijklmnopqrstuvwxyzz'

    // when
    const result = solution(s)

    // then
    assert.deepStrictEqual('a', result)
    done();
  });

  it('test 5', function (done) {
    // given
    const s = 'aabcdefghijklmnopqrstuvwxyz'

    // when
    const result = solution(s)

    // then
    assert.deepStrictEqual('a', result)
    done();
  });

  it('test 6', function (done) {
    // given
    const s = 'pvdfpgbnvx'

    // when
    const result = solution(s)

    // then
    assert.deepStrictEqual('p', result)
    done();
  });
})

File: 11기 코딩테스트_230308/4회차 코딩테스트/FirstAppearTwice/FirstAppearTwice_test.py
import unittest

from src.com.luckydave.algorithm.fastcampus.easy.FirstAppearTwice.FirstAppearTwice import solution


class CustomTestCase(unittest.TestCase):

    def test_1(self):
        # given
        s = 'google'

        # when
        result = solution(s)

        # then
        self.assertEqual('o', result)

    def test_2(self):
        # given
        s = 'naver'

        # when
        result = solution(s)

        # then
        self.assertEqual('', result)

    def test_3(self):
        # given
        s = 'abcdefghijklmnopqrstuvwxyzz'

        # when
        result = solution(s)

        # then
        self.assertEqual('z', result)

    def test_4(self):
        # given
        s = 'aabcdefghijklmnopqrstuvwxyzz'

        # when
        result = solution(s)

        # then
        self.assertEqual('a', result)

    def test_5(self):
        # given
        s = 'aabcdefghijklmnopqrstuvwxyz'

        # when
        result = solution(s)

        # then
        self.assertEqual('a', result)

    def test_6(self):
        # given
        s = 'pvdfpgbnvx'

        # when
        result = solution(s)

        # then
        self.assertEqual('p', result)


if __name__ == '__main__':
    unittest.main()
